ci/scripts/java_jni_manylinux_build.sh

```diff
@@ -45,6 +45,7 @@ export ARROW_ORC
 : ${ARROW_PLASMA:=ON}
 export ARROW_PLASMA
 : ${ARROW_S3:=ON}
+: ${ARROW_SUBSTRAIT:=ON}
```

Comment:
This has to always match ARROW_DATASET so a separate variable won't make sense

Reply:
changed